### DISCUSSION TITLE
Does enqueued to processing depends on polling interval?

### DISCUSSION THREAD

**Discussion by ap-sree** [jobrunr/jobrunr#1045] [Q&A]
Current config - 
Background server polling time - 10 mins
Recurring jobs - 3

Few times I need the trigger the job manually Run Now (via dashboard) or Move to enqueued via programmatically(From backend process).  Enqueued to Processing taking more time. Is this because of the polling time? 

And also the Recurring jobs running 10 mins earlier (most of time) than the scheduled time

**Comment by auloin** [comment]
> Few times I need the trigger the job manually Run Now (via dashboard) or Move to enqueued via programmatically(From backend process). Enqueued to Processing taking more time. Is this because of the polling time?

Indeed. The workers retrieve and process enqueued jobs every poll interval.

> And also the Recurring jobs running 10 mins earlier (most of time) than the scheduled time

This is because of your polling interval as noted in the docs https://www.jobrunr.io/en/documentation/background-methods/scheduling-jobs/ :

> whenever JobRunr fetches all the jobs that are scheduled and need to be executed, it fetches all jobs that need to happen in the next poll interval and enqueues them immediately.

Why do you need such a large polling interval?

Note that with JobRunr Pro, there is real-time scheduling and enqueued jobs are processed immediately as long as workers are free (independent of the configured poll interval), https://www.jobrunr.io/en/documentation/pro/real-time-scheduling/.
